source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -17,6 +18,17 @@ namespace Extensions {
 namespace HttpFilters {
 namespace HeaderToMetadataFilter {
 
+Rule::Rule(const std::string& header, const ProtoRule& rule) : header_(header), rule_(rule) {
+  if (rule.has_on_header_present()) {
+    const auto& on_present = rule.on_header_present();
+    if (on_present.has_regex_value_rewrite()) {
```

Comment:
you can collapse these if-blocks due to the way defaults are handled: `if (rule.on_header_present().has_regex_value_rewrite())` is safe